examples/co-group/src/main/java/com/hazelcast/jet/examples/cogroup/BatchCoGroup.java

```diff
@@ -0,0 +1,221 @@
+        BatchStage<Entry<Integer, Tuple3<List<PageVisit>, List<AddToCart>, List<Payment>>>> coGrouped =
+                pageVisits.aggregate3(toList(), addToCarts, toList(), payments, toList());
+
+        // Store the results in the output map
+        coGrouped.drainTo(Sinks.map(RESULT));
+        return p;
+    }
+
+    private static Pipeline coGroupBuild() {
+        Pipeline p = Pipeline.create();
+
+        // Create three source streams
+        BatchStageWithKey<PageVisit, Integer> pageVisits =
+                p.drawFrom(Sources.<PageVisit>list(PAGE_VISIT))
+                 .groupingKey(pageVisit -> pageVisit.userId());
+        BatchStageWithKey<AddToCart, Integer> addToCarts =
+                p.drawFrom(Sources.<AddToCart>list(ADD_TO_CART))
+                 .groupingKey(addToCart -> addToCart.userId());
+        BatchStageWithKey<Payment, Integer> payments =
+                p.drawFrom(Sources.<Payment>list(PAYMENT))
+                 .groupingKey(payment -> payment.userId());
+
+        // Obtain a builder object for the co-group transform
+        GroupAggregateBuilder<Integer, List<PageVisit>> builder = pageVisits.aggregateBuilder(toList());
+        Tag<List<PageVisit>> visitTag = builder.tag0();
+
+        // Add the co-grouped streams to the builder. Here we add just two, but
+        // you could add any number of them.
+        Tag<List<AddToCart>> cartTag = builder.add(addToCarts, toList());
+        Tag<List<Payment>> payTag = builder.add(payments, toList());
+
+        // Build the co-group transform. The aggregate operation collects all the
+        // stream items into ItemsByTag. We transform it into a 3-tuple of lists.
+        BatchStage<Entry<Integer, Tuple3<List<PageVisit>, List<AddToCart>, List<Payment>>>> coGrouped = builder
+                .build()
+                .map(keyAndVals -> {
+                    ItemsByTag ibt = keyAndVals.getValue();
+                    return entry(keyAndVals.getKey(), tuple3(ibt.get(visitTag), ibt.get(cartTag), ibt.get(payTag)));
+                });
+
+        // Store the results in the output map
+        coGrouped.drainTo(Sinks.map(RESULT));
+
+        return p;
+    }
+
+    public static void main(String[] args) {
+        System.setProperty("hazelcast.logging.type", "log4j");
+        JetInstance jet = Jet.newJetInstance();
+        Jet.newJetInstance();
+        new BatchCoGroup(jet).go();
+    }
+
+    private void go() {
+        prepareSampleData();
+        try {
+            jet.newJob(coGroupDirect()).join();
+            validateCoGroupResults();
+
+            jet.getMap(RESULT).clear();
+
+            jet.newJob(coGroupBuild()).join();
+            validateCoGroupResults();
+        } finally {
+            Jet.shutdownAll();
+        }
+    }
+
+    private void validateCoGroupResults() {
+        IMap<Integer, Tuple3<List<PageVisit>, List<AddToCart>, List<Payment>>> result = jet.getMap(RESULT);
+        printImap(result);
```

Comment:
The result looks like this:
```
result:
12->([PageVisit{3}, PageVisit{4}], [AddToCart{23}, AddToCart{24}], [Payment{33}, Payment{34}])
11->([PageVisit{1}, PageVisit{2}], [AddToCart{21}, AddToCart{22}], [Payment{31}, Payment{32}])
```
Would be good if the `Event` objects would print their `userId`s too, because then we could see that the result is correct.